mender-update/update_module/v3/update_module.hpp

```diff
@@ -161,12 +159,14 @@ class UpdateModule {
 	void StartDownloadToFile();
 
 	context::MenderContext &ctx_;
-	artifact::Payload &payload_;
-	artifact::PayloadHeaderView &payload_meta_data_;
 	string update_module_path_;
 	string update_module_workdir_;
 
-	struct {
+	class DownloadData {
+	public:
```

Comment:
What is a class with no function members and public-only members good for? To me it's just confusing that it's not a `struct`.

Also, this change seems to be a bit unrelated to the commit message.

Reply:
> What is a class with no function members and public-only members good for? To me it's just confusing that it's not a `struct`.

Hmm, yes, I see your point. This is a bit of a corner case. Even though the members are public in this class, they are private as seen from the outside, because of the outer `private` section. It's just that the implementation is handled by the outer class.

Also, it does have a constructor, which is non-trivial (though in the smallest possible sense).

I don't know, I don't have a strong leaning. It seems marginally more correct to me to use `class` if it has at least one function (the constructor), but I could go either way.

> Also, this change seems to be a bit unrelated to the commit message.

Well, it is related since it triggered the introduction of a constructor. But indeed, it's not immediately obvious, so I'll add a note about that in the message.

Reply:
> Which do you prefer?
I prefer `struct` for "passive" objects.

> I'll add a note about that in the message.
Thanks!

Reply:
Alright, done!

That means I also removed the message I just added, since now there is no change to that part.